modules/react/switch/lib/Switch.tsx

```diff
@@ -42,123 +34,142 @@ export interface SwitchProps extends Themeable {
   error?: ErrorType;
 }
 
-const circleSize = space.xs;
-const switchWidth = space.l;
-const switchHeight = space.s;
-const switchTapArea = space.l;
-const translateLength = space.s;
-
-const SwitchContainer = styled('div')({
-  position: 'relative',
-  height: space.m,
-  width: switchTapArea,
+const switchContainerStencil = createStencil({
+  base: {
+    position: 'relative',
+    height: system.space.x6,
+    width: system.space.x8,
+  },
 });
 
-const SwitchInput = styled('input')<SwitchProps & StyledType>(
-  {
-    position: 'absolute',
-    height: space.m,
-    width: switchTapArea,
-    margin: 0,
-    marginLeft: space.xxxs,
-    borderRadius: borderRadius.circle,
-    opacity: 0,
-    display: 'block',
+const SwitchContainer = createComponent('div')({
+  displayName: 'SwitchContainer',
+  Component: ({children, ...elemProps}, ref, Element) => {
+    return (
+      <Element ref={ref} {...mergeStyles(elemProps, switchContainerStencil())}>
+        {children}
+      </Element>
+    );
   },
-  ({disabled}) => ({
-    cursor: disabled ? 'not-allowed' : 'pointer',
-  }),
-  ({error, theme}) => {
-    const errorColors = getErrorColors(error, theme);
-
-    if (errorColors.outer === errorColors.inner) {
-      errorColors.outer = 'transparent';
-    }
+});
 
-    const styles = {
-      '&:focus': {
-        outline: 'none',
+const switchInputStencil = createStencil({
+  base: {
+    display: 'flex',
+    position: 'absolute',
+    height: system.space.x6,
+    width: system.space.x8,
+    margin: system.space.zero,
+    marginLeft: system.space.x1,
+    borderRadius: system.shape.round,
+    opacity: system.space.zero,
```

Comment:
opacity can't be `system.space.zero` because it's not rem value, token refers to rem value